### PR TITLE
Rz endian docs

### PR DESCRIPTION
**Pull request by createyourpersonalaccount**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [X] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

I noticed that the https://github.com/rizinorg/rizin/blob/5211cdcfb06cc0740d568d54a2a160453a36869e/librz/include/rz_endian.h header is undocumented and decided to document the functions therein. I also inserted new `UT24_MAX` (and `MIN`) macros. I added some missing functions so that every endianness and bit combination exists (except for some middle-endian combinations which I wasn't sure about).  I also added a macro identifier in the preprocessor namespace in case the builtin swap functions did not exist for uniformity in behavior, and ability to document. A function parameter was renamed from `_dest` to `dest` to match what other functions call their parameter.

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/include/rz_endian.h [hunk 1]
```diff
@@ -242,27 +555,84 @@ static inline ut16 rz_read_at_le16(const void *src, size_t offset) {
...
+/**
+ * \brief Read a 24-bit value from \p src in little-endian order.
+ * \param src The pointer from which the value is read.
+ * \return The read 24-bit value.
+ * \attention If \p src is \c NULL then \c UT24_MAX is returned.
+ */
+static inline ut32 rz_read_le24(const void *src) {
+	if (!src) {
+		return UT24_MAX;
+	}
```
Comment (ret2libc) [inline librz/include/rz_endian.h]: I wonder whether we should just return UT32_MAX in these cases... @XVilka @wargio WDYT?

**Review by ret2libc — COMMENTED** [review]
Wow thanks for the documentation, it's great!

File: librz/include/rz_endian.h (hunk 1 above)
Comment (XVilka) [reply librz/include/rz_endian.h]: UT32_MAX would be more appropriate, yes.

File: librz/include/rz_endian.h [hunk 2]
```diff
@@ -646,24 +1581,192 @@ static inline void rz_write_ble(void *dst, ut64 val, bool big_endian, int size)
...
+ * assert(rz_swap_ut16(x) == result);
+ * \endcode
+ */
 #if HAVE___BUILTIN_BSWAP16
 #define rz_swap_ut16 __builtin_bswap16
 #else
 static inline ut16 rz_swap_ut16(ut16 val) {
 	return (val << 8) | (val >> 8);
 }
+#define rz_swap_ut16 rz_swap_ut16
```
Comment (XVilka) [inline librz/include/rz_endian.h]: What is this for?

**Review by XVilka — CHANGES_REQUESTED** [review]
If the function is wrong, please provide a fix, also update/create a test for it, see `test/unit/test_endian.c`
I also recommend checking the assembly output with [Compiler Explorer](https://godbolt.org/) for x86 and ARM64, to make sure the updated version produces fast and short code, it's in the "hot" path.

File: librz/include/rz_endian.h (hunk 1 above)
Comment (createyourpersonalaccount) [reply librz/include/rz_endian.h]: Okay so it returns the maximum value that fits in the return type. I will correct this.

File: librz/include/rz_endian.h (hunk 2 above)
Comment (createyourpersonalaccount) [reply librz/include/rz_endian.h]: With Doxygen I document the `rz_swap_ut16` macro. I wasn't sure if the macro needed to be present or not, so I made sure it was always defined. Maybe I should just remove the extra `#define` and keep the Doxygen comment as it is?

Comment (XVilka) [reply librz/include/rz_endian.h]: Yes, just remove the macro

**Comment by createyourpersonalaccount** [comment]
@XVilka Why not use this?

```c
st16 rz_swap_st16(st16 val) { return (st16)rz_swap_ut16((ut16)val); }
```

**Comment by createyourpersonalaccount** [comment]
@XVilka I decided to remove the signed swap functions, they were unused.

**Review by XVilka — CHANGES_REQUESTED** [review]
With MSVC:
```
../test/unit/test_endian.c(428): error C2059: syntax error: ','
../test/unit/test_endian.c(429): error C2059: syntax error: ','
../test/unit/test_endian.c(430): error C2059: syntax error: ','
../test/unit/test_endian.c(431): error C2059: syntax error: ','
../test/unit/test_endian.c(432): error C2059: syntax error: ','
../test/unit/test_endian.c(434): error C2059: syntax error: ','
../test/unit/test_endian.c(435): error C2059: syntax error: ','
../test/unit/test_endian.c(436): error C2059: syntax error: ','
../test/unit/test_endian.c(437): error C2059: syntax error: ','
../test/unit/test_endian.c(438): error C2059: syntax error: ','
../test/unit/test_endian.c(440): error C2059: syntax error: ','
../test/unit/test_endian.c(441): error C2059: syntax error: ','
../test/unit/test_endian.c(442): error C2059: syntax error: ','
[1862/2031] Compiling C object test/unit/test_ebcdic.exe.p/test_ebcdic.c.obj
[1863/2031] Compiling C object test/unit/test_event.exe.p/test_event.c.obj
../test/unit/test_event.c(42): warning C4312: 'type cast': conve
```

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]
wow <3

**Pull request merged**